### PR TITLE
Add rough draft of Fedora 3-related ADR

### PR DESCRIPTION
**Pull request by mjgiarlo**

### CODE REVIEW

File: 0001-migrate-off-fedora-3.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+### Maintain Fedora 3
+
+* Bad, see decision drivers above.
+* Bad, lack of resources and time to absorb Fedora 3 into our portfolio
+* Bad, lack of expertise with Java development
+* Bad, completely out of step with community trends
+
+### Replace Fedora 3 as a metadata store with PostgreSQL
+
```
Comment (jermnelson) [inline 0001-migrate-off-fedora-3.md]: Can the *Good* , *Bad*, *Neutral* be better grouped? Like having  a (Good, etc) subheading with each bullet item underneath that heading? Another option might be a markdown table (but that introduces more complexity in the markdown)

File: 0001-migrate-off-fedora-3.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+* Good, Postgres is a performant, scalable, well-tested, open-source solution in wide use
+* Good, Postgres is the most common replacement for Fedora in the Samvera community, now that Valkyrie is catching on
+* Good, Postgres supports both schemaless data (JSON) and schema-based data, with built-in indexing, relationships, and transactions
+* Neutral, Postgres incentivizes changing our metadata model to be based on JSON (see also: COCINA data model)
+* Bad, Fedora functionality missing in Postgres includes "object" versioning (and services pertaining to binaries, which DOR does not use)
+* Good, Should we decide to make more use of cloud-based hosting services, Postgres is easier to support and most cloud providers have a native Postgres-backed service (for cost reduction)
+* Good, Peer institutions such as Princeton, Northwestern, and BPL are using Postgres
+
```
Comment (peetucket) [inline 0001-migrate-off-fedora-3.md]: Pro: assuming we stay with Rails, Postgres works well with ActiveRecord and other rails patterns.

File: 0001-migrate-off-fedora-3.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+### Negative Consequences <!-- optional -->
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* ...
+
+## Pros and Cons of the Options <!-- optional -->
+
+### Do Nothing
+
+* Bad, see decision drivers above.
```
Comment (peetucket) [inline 0001-migrate-off-fedora-3.md]: Pro: less work (for now, until it falls apart)  -- not saying i agree with this, but good to acknowledge

File: 0001-migrate-off-fedora-3.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+Fedora 3 is unsupported and has been unsupported for four years; it is unlikely to be supported in the foreseeable future. It also requires a difficult-to-support version of the Java Virtual Machine. And yet, Fedora 3 is the cornerstone of our management "repository," in which all SDR content is managed and from which said content flows to access and preservation environments. At the same time, there is a dwindling number of organizations in the cultural heritage community who are still using Fedora 3.
+
+## Decision Drivers <!-- optional -->
+
+* Fedora 3 is unsupported and unlikely to be supported
+* Fedora 3 will be harder to install on newer operating systems
+* The Fedora 3 data model is not inherently validatable
+* The Fedora 3 community is disappearing, so we are increasingly going it alone
+* Fedora 3 is a critical piece of SDR infrastructure and represents an enormous risk
+* Samvera software that supports Fedora 3 is outdated and maintained/supported only through our own efforts, preventing us from using mainstream Samvera software
```
Comment (justinlittman) [inline 0001-migrate-off-fedora-3.md]: We have unverified concerns about the scalability of Fedora 3?

File: 0001-migrate-off-fedora-3.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+* Good, Postgres is the most common replacement for Fedora in the Samvera community, now that Valkyrie is catching on
+* Good, Postgres supports both schemaless data (JSON) and schema-based data, with built-in indexing, relationships, and transactions
+* Neutral, Postgres incentivizes changing our metadata model to be based on JSON (see also: COCINA data model)
+* Bad, Fedora functionality missing in Postgres includes "object" versioning (and services pertaining to binaries, which DOR does not use)
+* Good, Should we decide to make more use of cloud-based hosting services, Postgres is easier to support and most cloud providers have a native Postgres-backed service (for cost reduction)
+* Good, Peer institutions such as Princeton, Northwestern, and BPL are using Postgres
+
+### Replace Fedora 3 with latest Fedora version
+
+* Neutral, Fedora 4+ incentivizes changing our metadata model to be based on RDF
```
Comment (justinlittman) [inline 0001-migrate-off-fedora-3.md]: Bad.

File: 0001-migrate-off-fedora-3.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+* Bad, Fedora has many fewer users and maintainers than Postgres
+* Bad, There is a significantly smaller set of tools that interoperate with Fedora (and RDF) compared with Postgres (and JSON and/or relational data)
+* Bad, Fedora has proven to be challenging for new developers to become conversant with, especially compared with relational databases like Postgres
+* Bad, Fedora lacks transactions, necessitating quasi-transactional functionality at higher layers, which means more code to maintain and opportunities for errors
+* Good, Continuing to use Fedora keeps us aligned with the broader repository community (Islandora/Fedora), though perhaps not with Samvera community trends
+* Bad, Fedora 4+ still has scalability concerns, and we have no reason to believe Fedora 5 or 6 are/will be any better
```
Comment (justinlittman) [inline 0001-migrate-off-fedora-3.md]: Also, unclear how similar future versions of Fedora will be to Fedorta 4.

File: 0001-migrate-off-fedora-3.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+* Bad, see decision drivers above.
+
+### Maintain Fedora 3
+
+* Bad, see decision drivers above.
+* Bad, lack of resources and time to absorb Fedora 3 into our portfolio
+* Bad, lack of expertise with Java development
+* Bad, completely out of step with community trends
+
+### Replace Fedora 3 as a metadata store with PostgreSQL
```
Comment (justinlittman) [inline 0001-migrate-off-fedora-3.md]: Is this proposal "replace with PostgreSQL" or "replace with Valkyrie with a PostgreSQL backend" or "replace with Valkyrie and some backend"?

Comment (mjgiarlo) [reply 0001-migrate-off-fedora-3.md]: @justinlittman Do you think there are Valkyrie assumptions baked into this section?

File: 0001-migrate-off-fedora-3.md (hunk 6 above)
Comment (mjgiarlo) [reply 0001-migrate-off-fedora-3.md]: Added.

File: 0001-migrate-off-fedora-3.md (hunk 1 above)
Comment (mjgiarlo) [reply 0001-migrate-off-fedora-3.md]: Changed (to use sub-bullets). Thanks, looks much better now!

**Comment by mjgiarlo** [comment]
> Suggestion: use "Pro/Con" language in bullet points instead of "Good/bad". Also suggest grouping them together (so subgroups under each main category for "pro" (good), "con" (bad) and "note" (neutral).

Done and done, @peetucket. Thx!

File: 0001-migrate-off-fedora-3.md (hunk 2 above)
Comment (mjgiarlo) [reply 0001-migrate-off-fedora-3.md]: Added.

File: 0001-migrate-off-fedora-3.md (hunk 3 above)
Comment (mjgiarlo) [reply 0001-migrate-off-fedora-3.md]: Added.

File: 0001-migrate-off-fedora-3.md (hunk 4 above)
Comment (mjgiarlo) [reply 0001-migrate-off-fedora-3.md]: Added.

File: 0001-migrate-off-fedora-3.md (hunk 5 above)
Comment (mjgiarlo) [reply 0001-migrate-off-fedora-3.md]: Added.

File: 0001-migrate-off-fedora-3.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+* We have (unverified) concerns about the scalability of Fedora 3
+
+## Considered Options <!-- required -->
+
+* Do nothing
+* Keep using Fedora 3 & DLSS Infrastructure Team takes over maintenance of Fedora 3 and Fedora 3-supporting software in Samvera community
+* Replace Fedora 3 as a metadata store with PostgreSQL
+* Replace Fedora 3 with latest Fedora version
+
+## Decision Outcome <!-- required -->
```
Comment (ndushay) [inline 0001-migrate-off-fedora-3.md]: Infra team recommendation for heading?

File: 0001-migrate-off-fedora-3.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  * Julian Morley
+  * Andrew Berger
+  * Vivian Wong
+  * Tom Cramer
+  * Hannah Frost
+  * Infrastructure Team
+* Author(s):
+  * Justin Coyne
+  * Mike Giarlo
+  * Peter Mangiafico
```
Comment (ndushay) [inline 0001-migrate-off-fedora-3.md]: Add "Approvers" section?

**Pull request merged**